### PR TITLE
call async datastore from modbus server execute flow

### PR DESCRIPTION
**Pull request by ilkka-ollakka**
Take into use of async_getValues and async_setValues in modbus-server message handling, allowing datastore (mainly remote datastore) to use async flow.

Originally split from https://github.com/pymodbus-dev/pymodbus/pull/2127

### CODE REVIEW

**Comment by janiversen** [comment]
A PR is independent, we do not merge/split PRs. 

If a PR is depending on other PRs then please mark it as draft, so nobody wastes time.

**Comment by janiversen** [comment]
I have lost the overview...you have 3 open PR`s, is any of them ready to be merged (all comments resolved) ??

Seems you do not allow maintainers to update the PR, so I cannot rebase the 3 PRs, please do a rebase on the one where you want a final review.

**Comment by ilkka-ollakka** [comment]
> I have lost the overview...you have 3 open PR`s, is any of them ready to be merged (all comments resolved) ??

This one is only one that depends on those 2 other PR's, to my understanding all comments have been reacted.
 
> Seems you do not allow maintainers to update the PR, so I cannot rebase the 3 PRs, please do a rebase on the one where you want a final review.

All are now rebased

**Comment by janiversen** [comment]
Is this ready for review ?

**Review by janiversen — CHANGES_REQUESTED** [review]
It might be that remote returns a value in setValues, but we do not expect a value !

Apart from that the code is currently flawed, because not all getValues() calls check for ExceptionResponse, but this is outside this PR.

File: pymodbus/pdu/bit_write_message.py
Comment (ilkka-ollakka) [reply pymodbus/pdu/bit_write_message.py]: fixed

File: pymodbus/pdu/register_read_message.py
Comment (ilkka-ollakka) [reply pymodbus/pdu/register_read_message.py]: fixed

File: pymodbus/pdu/register_write_message.py
Comment (ilkka-ollakka) [reply pymodbus/pdu/register_write_message.py]: fixed

File: pymodbus/pdu/register_write_message.py
Comment (ilkka-ollakka) [reply pymodbus/pdu/register_write_message.py]: fixed

File: pymodbus/pdu/register_write_message.py
Comment (ilkka-ollakka) [reply pymodbus/pdu/register_write_message.py]: fixed

**Pull request merged**
